lib/service/Service.ts

```diff
@@ -349,6 +350,44 @@ class Service {
     return this.pool.listPeers();
   }
 
+  /**
+   * Get all trades, swaps and order status related to an order.
+   */
+  public listOrderHistory = async (args: { orderId: string }): Promise<{
+    trades: TradeInstance[], swapDeals: SwapDealInstance[], status: OrderStatus}> => {
+    const order = await this.orderBook.getOrder(args.orderId);
+
+    if (order === null) {
+      throw errors.PAIRID_NON_EXISTENT();
+    }
+
+    const trades = await this.orderBook.getTradesByOrderId(order.id, true);
+    const swapDeals = await this.swaps.getDeals(true);
+
+    const isInOrderBook = this.orderBook.getOwnOrder(order.id, order.pairId);
+    let quantity = 0;
+    let status: OrderStatus = OrderStatus.OPEN;
+    trades.forEach((trade) => {
+      quantity = quantity + trade.quantity;
+    });
+
+    if (!isInOrderBook && quantity === order.initialQuantity) {
+      status = OrderStatus.FILLED;
+    } else if (!isInOrderBook && trades.length === 0) {
+      status = OrderStatus.CANCELLED;
+    } else if (isInOrderBook && trades.length === 0) {
+      status = OrderStatus.OPEN;
```

Comment:
`status` is `OPEN` when it's initialized above, I figure we can leave this condition out?